src/pages/index.js

```diff
@@ -1,107 +1,244 @@
+  const deleteConfirmationPopup = new PopupWithConfirmation(
+    "#delete-confirmation-modal",
+    (cardId) => {
+      api
+        .deleteCard(cardId)
+        .then(() => {
+          const cardElement = document.querySelector(`[data-id='${cardId}']`);
+          if (cardElement) {
+            cardElement.remove();
+          }
+          console.log("Card deleted successfully.");
+        })
+        .catch((error) => {
+          console.error("Error deleting card:", error);
+        });
+    }
+  );
+
   const profileFormValidator = new FormValidator(settings, profileEditForm);
   const addCardFormValidator = new FormValidator(settings, addCardForm);
-  // Enable form validation
   profileFormValidator.enableValidation();
   addCardFormValidator.enableValidation();
 
-  /* ---------------------------- Handlers ----------------------------- */
-  // Function to create a new Card instance
-  function createCard(cardData) {
-    const card = new Card(cardData, "#card-template", handleImageClick);
-    return card.getView();
+  // Select avatar form and initialize its validator
+  const avatarForm = document.querySelector("#profile-avatar-form");
+  const avatarFormValidator = new FormValidator(settings, avatarForm);
+  avatarFormValidator.enableValidation(); // Now correctly initialized
+
+  // Function to create a card
+  function createCard(cardData, api) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      api,
+      deleteConfirmationPopup
+    ); // Pass api to Card
+    const cardElement = card.getView(); // Get the card view correctly
+    cardElement.dataset.id = cardData._id;
+    // Attach event listener for the delete button after the card has been created
+    cardElement
+      .querySelector(".card__delete-button")
+      .addEventListener("click", () => {
+        deleteConfirmationPopup.open(cardData._id);
+      });
+
+    return cardElement; // Return the correctly initialized card element
   }
 
-  // Handle image clicks to open the image preview popup
   function handleImageClick(data) {
-    popupWithImage.open(data); // Open image popup with data (name and link)
+    popupWithImage.open(data);
   }
 
-  // Handle opening the profile edit modal and populating it with current user data
   profileEditButton.addEventListener("click", () => {
     const userData = userInfo.getUserInfo();
     nameInput.value = userData.name;
     descriptionInput.value = userData.job;
-    profileEditPopup.open(); // Open the modal
-    profileFormValidator.resetValidation(); // Reset form validation for profile edit popup
+    profileEditPopup.open();
+    profileFormValidator.resetValidation();
   });
 
-  // Handle opening the modal for adding a new card
   addButton.addEventListener("click", () => {
-    addCardFormValidator.disableButton(); // Only disable button when opening add card popup
-    addCardPopup.open(); // Open the modal
+    addCardFormValidator.disableButton();
```

Comment:
`addCardFormValidator.disableButton();` is not needed here

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.